ibu-imager/seedcleanup/seedcleanup.go

```diff
@@ -0,0 +1,174 @@
+limitations under the License.
+*/
+
+package seedcleanup
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/openshift-kni/lifecycle-agent/ibu-imager/common"
+	"github.com/openshift-kni/lifecycle-agent/ibu-imager/ops"
+	"github.com/openshift-kni/lifecycle-agent/internal/clusterconfig"
+	"github.com/sirupsen/logrus"
+)
+
+// SeedCleaner handles cleanup operations after creating a seed image, removing temporary files
+// and executing additional cleanup steps as needed.
+type SeedCleaner struct {
+	log                  *logrus.Logger
+	ops                  ops.Ops
+	backupDir            string
+	containerRegistry    string
+	authFile             string
+	recertContainerImage string
+	recertSkipValidation bool
+}
+
+func NewSeedCleaner(log *logrus.Logger, ops ops.Ops, backupDir,
+	containerRegistry, authFile, recertContainerImage string, recertSkipValidation bool) *SeedCleaner {
+
+	return &SeedCleaner{
+		log:                  log,
+		ops:                  ops,
+		backupDir:            backupDir,
+		containerRegistry:    containerRegistry,
+		authFile:             authFile,
+		recertContainerImage: recertContainerImage,
+		recertSkipValidation: recertSkipValidation,
+	}
+}
+
+// CleanupSeedCluster comprises the Imager workflow for cleanup operations after creating a seed image
+// out of an SNO cluster.
+func (s *SeedCleaner) CleanupSeedCluster() error {
+	s.log.Println("Cleaning up seed cluster")
+
+	if err := s.cleanupBackupDir(); err != nil {
+		return err
+	}
+
+	if err := s.cleanupConfigurationFiles(); err != nil {
+		return err
+	}
+
+	if s.recertSkipValidation {
+		s.log.Info("Skipping restoring certificates in etcd via recert tool.")
+	} else {
+		if err := s.restoreCertificates(); err != nil {
+			return err
+		}
+	}
+
+	if err := s.restoreCluster(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *SeedCleaner) cleanupBackupDir() error {
+	if err := os.RemoveAll(s.backupDir); err != nil {
+		return fmt.Errorf("error removing %s: %w", s.backupDir, err)
+	}
+
+	if err := os.Remove(common.BackupChecksDir + "/container_list.done"); err != nil && !os.IsNotExist(err) {
+		s.log.Errorf("error removing %s: %v", common.BackupChecksDir+"/container_list.done", err)
+	}
+
+	s.log.Info("Backup data removed successfully.")
+	return nil
+}
+
+func (s *SeedCleaner) cleanupConfigurationFiles() error {
+	for _, systemdUnit := range common.SystemdUnitsToDisable {
+		if _, err := s.ops.SystemctlAction("disable", systemdUnit); err != nil {
+			s.log.Errorf("error disabling %s unit: %v", systemdUnit, err)
+		}
+	}
+	s.log.Info("Preparation systemd units disabled successfully.")
+
+	for _, file := range common.FilesToRemove {
+		if err := os.Remove(file); err != nil && !os.IsNotExist(err) {
+			s.log.Errorf("error removing %s: %v", file, err)
+		}
+	}
+	s.log.Info("Preparation systemd scripts removed successfully.")
+
+	return nil
+}
+
+func (s *SeedCleaner) restoreCertificates() error {
```

Comment:
nit: Recert is not just about certs (despite the name)

```suggestion
func (s *SeedCleaner) restoreOriginalSeedCrypto() error {
```